tests/testthat/test-merge_sce_list.R

```diff
@@ -397,59 +469,82 @@ test_that("merging SCEs with same altExp features works as expected, with altexp
     unname()
   expect_equal(colnames(merged_altexp), expected_colnames)
 
-})
+  # Check colData columns
+  expected_coldata <- c(
+    "sum",
+    "detected",
+    "altexps_adt_sum",
+    "altexps_adt_detected",
+    "altexps_adt_percent",
+    batch_column,
+    "cell_id"
+  )
 
+  expect_true(
+    setequal(
+      colnames(colData(merged_sce)),
+      expected_coldata
+    )
+  )
+})
 
 
 
-test_that("merging SCEs with different altExp features works as expected, with altexps", {
 
+test_that("merging SCEs with 1 altexp but different features fails as expected, with altexps", {
   # keep only the first 3 features from the first SCE
-  altExp(sce_list_with_altexp[[1]]) <- altExp(sce_list_with_altexp[[1]])[1:3]
+  altExp(sce_list_with_altexp[[1]]) <- altExp(sce_list_with_altexp[[1]])[1:3, ]
 
 
-  merged_sce <- merge_sce_list(
-    sce_list_with_altexp,
-    batch_column = batch_column,
-    # "total" should get removed
-    retain_coldata_cols = retain_coldata_cols,
-    # this row name should not be modified:
-    preserve_rowdata_cols = c("gene_names")
+  expect_error(
+    merge_sce_list(
+      sce_list_with_altexp,
+      batch_column = batch_column,
+      # "total" should get removed
+      retain_coldata_cols = retain_coldata_cols,
+      # this row name should not be modified:
+      preserve_rowdata_cols = c("gene_names")
+    )
   )
+})
 
-  merged_altexp <- altExp(merged_sce)
-
-
-  expect_true(altExpNames(merged_sce) == altexp_name)
-  expect_equal(dim(merged_altexp), c(num_altexp_features, total_cells))
-  expect_equal(rownames(merged_altexp), full_altexp_features)
 
-  expected_colnames <- sce_list_with_altexp |>
-    purrr::imap(
-      \(sce, sce_name) glue::glue("{sce_name}-{colnames(sce)}")
-    ) |>
-    unlist() |>
-    unname()
-  expect_equal(colnames(merged_altexp), expected_colnames)
 
-})
 
+test_that("merging SCEs where 1 altExp is missing works as expected, with altexps", {
+  sce_list_with_altexp[["sce4"]] <- removeAltExps(sce_list_with_altexp[[1]])
 
-test_that("merging SCEs with altExps works as expected when include_altexps = FALSE", {
+  # from cbind docs:
+  # The colnames in colData(SummarizedExperiment) must match or an error is thrown.
+  # Duplicate columns of rowData(SummarizedExperiment) must contain the same data.
 
   merged_sce <- merge_sce_list(
     sce_list_with_altexp,
     batch_column = batch_column,
     # "total" should get removed
     retain_coldata_cols = retain_coldata_cols,
     # this row name should not be modified:
-    preserve_rowdata_cols = c("gene_names"),
-    include_altexp = FALSE
+    preserve_rowdata_cols = c("gene_names")
   )
 
-  # there should not be any altExps
-  expect_length(altExpNames(merged_sce), 0)
+  expect_equal(altExpNames(merged_sce), "adt")
```

Comment:
Let's add in a few more tests here:
Check that the number of columns is what we expect, for the main and altexps
Check that the colData columns are still as expected?
(same checks as when all were present, basically)

Check that the missing values is all `NA`, perhaps? (Pull out the values where the library is "sce4"?)